playbooks/roles/ambari-blueprint/templates/blueprint_dynamic.j2

```diff
@@ -376,6 +376,20 @@
         "nifi.sensitive.props.key": "{{ nifi_security_options.sensitive_props_key }}"
       }
     },
+    { 
+      "nifi-ambari-ssl-config" : {
+        "nifi.initial.admin.identity": "{{ nifi_initial_admin_user }}",
+        "nifi.node.ssl.isenabled": "true",
+        "nifi.security.keystore": "{{ keystore }}",
+        "nifi.security.keystorePasswd": "{{ keystore_pass }}",
+        "nifi.security.keystoreType": "JKS",
+        "nifi.security.truststore": "{{ truststore }}",
+        "nifi.security.truststorePasswd": "{{ truststore_pass }}",
+        "nifi.security.truststoreType": "JKS",
+        "nifi.toolkit.dn.suffix": "{{ nifi_dn_suffix }}",
+        "content": "{{ nifi_ambari_ssl_config_content }}"
+      }
+    },
```

Comment:
We want to maintain compatibility with the upstream Hortonworks version, so we should set the default values to the empty string for all these properties because other users may not have variables like `nifi_initial_admin_user` defined in their Ansible playbooks.

I defined 3 new variables: 
- `enable_nifi_ssl` (set default to false here because the README for the upstream repo says NiFi SSL is not supported, but I imagine it'll be set to true in our environment file)
- `nifi_security_keystore_type`
- `nifi_security_truststore_type`

Here's an example of how to do this:
`
{ 
      "nifi-ambari-ssl-config" : {
        "nifi.initial.admin.identity": "{{ nifi_initial_admin_user | default('') }}",
        "nifi.node.ssl.isenabled": "{{ enable_nifi_ssl | default(false) }}",
        "nifi.security.keystore": "{{ keystore | default('')}}",
        "nifi.security.keystorePasswd": "{{ keystore_pass | default('') }}",
        "nifi.security.keystoreType": "{{ nifi_security_keystore_type | default('') }}",
        "nifi.security.truststore": "{{ truststore | default('') }}",
        "nifi.security.truststorePasswd": "{{ truststore_pass | default('') }}",
        "nifi.security.truststoreType": "{{ nifi_security_truststore_type | default('') }}",
        "nifi.toolkit.dn.suffix": "{{ nifi_dn_suffix | default('') }}",
        "content": "{{ nifi_ambari_ssl_config_content | default('') }}"
      }
    },
`